src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,206 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> first;
+    private Node<T> last;
+    private int size = 0;
+    private  int midIndex = size / 2;
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            addFirstNode(value);
+        } else {
+            addNoteToEnd(value);
+        }
     }
 
     @Override
     public void add(T value, int index) {
+        checkElementIndexForAdd(index);
+        if (size == 0) {
+            addFirstNode(value);
+        } else if (index == 0) {
+            addNodeToStart(value);
+        } else if (index == size) {
+            addNoteToEnd(value);
+        } else {
+            Node<T> currentNode = findNodeByIndex(index);
+            Node<T> newNode = new Node<>(currentNode.prev, value, currentNode.next);
```

Comment:
When creating a new `Node<T>`, pass all the fields to the `Node` constructor. Also, the `next` field should be `currentNode`, not `currentNode.next`.